src/sign/index.ts

```diff
@@ -1,14 +1,43 @@
 import { createHmac } from "crypto";
 
-export function sign(secret: string, payload: string | object): string {
+export enum Algorithm {
+  SHA1 = "sha1",
+  SHA256 = "sha256",
+}
+
+type SignOptions = {
+  secret: string;
+  algorithm?: string;
+};
+
+export function sign(
+  options: SignOptions | string,
+  payload: string | object
+): string {
+  let secret;
+  let algorithm;
+
+  if (typeof options === "string") {
+    secret = options;
+    algorithm = Algorithm.SHA1;
+  } else {
+    secret = options.secret;
+    algorithm = options.algorithm ?? Algorithm.SHA1;
+  }
   // @ts-ignore throw friendly error message when required options are missing
   if (!secret || !payload) {
     throw new TypeError("secret & payload required");
   }
 
+  if (!Object.values(Algorithm).includes(algorithm as Algorithm)) {
+    throw new TypeError("algorithm should be 'sha1' or 'sha256'");
+  }
```

Comment:
Should be this something we control from `@octokit/webhooks.js` or we should pass whatever string the user enters and `createHmac()` should buble up the error?

---

In case we keep this `enum` approach, TypeScript question, is there a more elegant way to control internally if a `string` is one of the allowed values under an `enum` (but keep using `string` instead of `enum` as signature of the method)?

Reply:
I'd prefer to control it. GitHub only supports `sha1` and `sha256`, it makes no sense for us to support any other algorithm, and could be the source of a hard to debug problem. Throwing an error will make it obvious.

As a side note, I started to prefix error and log messages with the package name
```suggestion
  if (!Object.values(Algorithm).includes(algorithm as Algorithm)) {
    throw new TypeError("[@octokit/webhooks] algorithm should be 'sha1' or 'sha256'");
  }
```

We should do that in with all errors / log messages, but in a separate PR

Reply:
Created issue associated to this comment ;)

Reply:
> In case we keep this enum approach, TypeScript question, is there a more elegant way to control internally if a string is one of the allowed values under an enum (but keep using string instead of enum as signature of the method)?

@wolfy1339 TS question to the rescue! Any good idea on how to control this in a less hacky way? 😇